model-optimizer/extensions/middle/ReverseV2ToReverseSequence.py

```diff
@@ -32,29 +35,51 @@ def pattern():
+        if in_shape_rank == 1:
+            unsq_node = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
+                                                         {'name': reverse_name+"/Unsqueeze"})
+            reverse.in_port(0).get_source().connect(unsq_node.in_port(0))
+            new_in = unsq_node.out_port(0)
+            batch_axis = 0
+            # add 1 for newly added dimension
+            seq_axis = reverse['axis'] + 1 if reverse['axis'] >= 0 else reverse['axis']
+        else:
+            new_in = reverse.in_port(0).get_source()
+            seq_axis = reverse['axis']
+            batch_axis = 0 if seq_axis != 0 else 1
 
         # 1. For ReverseSequence 1-port input is seq_lengths => create this input node
-        seq_lengths = np.ones(input_data_shape[batch_axis]) * input_data_shape[seq_axis]
-
         reverse_name = reverse.soft_get('name',  reverse.id)
         rename_node(reverse, reverse_name + '/to_delete')
+
+        shape_node = Shape(graph, {'name': reverse_name + "/shape"}).create_node()
+        new_in.connect(shape_node.in_port(0))
+        seq_axis_node = node_to_get_shape_value_of_indices(shape_node, [seq_axis])
+        batch_node = node_to_get_shape_value_of_indices(shape_node, [batch_axis])
+        broadcast_node = Broadcast(graph, {'name': reverse_name + "/broadcast"}).create_node()
+        broadcast_node.in_port(0).connect(seq_axis_node.out_port(0))
+        broadcast_node.in_port(1).connect(batch_node.out_port(0))
+
         # 2. Create new ReverseSequence node and reconnect all inputs/outputs to it
-        reverse_sequence = create_op_node_with_second_input(graph, ReverseSequence, seq_lengths,
-                                                            {'name':  reverse_name, 'seq_axis': seq_axis,
-                                                             'batch_axis': batch_axis})
-        rename_node(reverse_sequence, reverse_name)
-        reverse.in_port(0).get_connection().set_destination(reverse_sequence.in_port(0))
-        reverse.out_port(0).get_connection().set_source(reverse_sequence.out_port(0))
+        reverse_sequence = ReverseSequence(graph, {'name':  reverse_name, 'seq_axis': seq_axis,
+                                                   'batch_axis': batch_axis}).create_node()
+        reverse_sequence.in_port(0).connect(new_in)
+        reverse_sequence.in_port(1).connect(broadcast_node.out_port(0))
+
+        # remove added dimension for rank = 1
+        if in_shape_rank == 1:
+            squeeze_node = create_op_node_with_second_input(graph, Squeeze, int64_array([0]),
+                                                            {'name': reverse_name + "/Squeeze"})
+            squeeze_node.in_port(0).connect(reverse_sequence.out_port(0))
+            reverse.out_port(0).get_connection().set_source(squeeze_node.out_port(0))
+        else:
+            reverse.out_port(0).get_connection().set_source(reverse_sequence.out_port(0))
 
```

Comment:
please preserve a node name. In some case you should move `reverse_name` for `Squeeze` node and for others it must be `ReverseSequence`

Reply:
fixed